Refactor AzoriusV1 contract to improve strategy retrieval logic

adamgall opened the pull request:
- Added error handling for invalid start addresses and zero counts in the `getStrategies` function.
- Updated the logic to correctly return the next strategy when not all strategies fit in the returned array.
- Enhanced tests to validate new error handling and pagination behavior.

These changes enhance the robustness and clarity of the strategy management functionality in the AzoriusV1 contract.

twodam commented on contracts/deployables/modules/AzoriusV1.sol:
```diff
@@ -269,22 +271,37 @@ contract AzoriusV1 is IAzoriusV1, GuardableModule, Version, UUPSUpgradeable {
         uint256 _count
     ) external view returns (address[] memory _strategies, address _next) {
+        if (
+            _startAddress != SENTINEL_STRATEGY &&
```
this check has been included in function `isStrategyEnabled`, is it necessary to write this again?
```solidity
function isStrategyEnabled(address _strategy) public view returns (bool) {
        return
            SENTINEL_STRATEGY != _strategy &&
            strategies[_strategy] != address(0);
    }
    ```

adamgall committed:
Refactor AzoriusV1 contract to improve strategy retrieval logic

- Removed the `AzoriusSetUp` event as it is no longer needed.
- Added error handling for invalid start addresses and zero counts in the `getStrategies` function.
- Updated the logic to correctly return the next strategy when not all strategies fit in the returned array.
- Enhanced tests to validate new error handling and pagination behavior.

These changes enhance the robustness and clarity of the strategy management functionality in the AzoriusV1 contract.

dacebt reviewed (approved):
Sweet, so with this change (noting), In the frontend we call this method in 3 places.

`DaoHierachyNode`, `useDeployAzorius` and `useVotingStrategiesAddresses` and currently none of these places catch any revert.

dacebt commented:
PS (minor). PR description says `Removed the AzoriusSetUp event as it is no longer needed.` which I don't think is true anymore
